### PR TITLE
implemented dao and services for hibernate practice v 1.0

### PR DESCRIPTION
**Pull request by CodeWebWeaver**

### CODE REVIEW

File: src/main/java/mate/academy/hibernate/relations/dao/impl/ActorDaoImpl.java [hunk 1]
```diff
@@ -12,11 +15,47 @@ public ActorDaoImpl(SessionFactory sessionFactory) {
...
+            }
+            throw new DataProcessingException("Error occurred "
+                    + "while getting actor by id: " + id
+                    + "\nConnection failed", e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
     }
```
Comment (keda-vlad) [inline src/main/java/mate/academy/hibernate/relations/dao/impl/ActorDaoImpl.java:60]: Common mistakes - "Use try with finally, where you use transaction and try with resources with Read operations." 
U don`t need transactions here at all, so just open session it try-with-resources

File: src/main/java/mate/academy/hibernate/relations/dao/impl/CountryDaoImpl.java [hunk 2]
```diff
@@ -12,11 +15,47 @@ public CountryDaoImpl(SessionFactory sessionFactory) {
...
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Error occurred "
+                    + "while getting country by id: " + id
+                    + "\nConnection failed", e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
```
Comment (keda-vlad) [inline src/main/java/mate/academy/hibernate/relations/dao/impl/CountryDaoImpl.java:59]: Common mistakes - "Use try with finally, where you use transaction and try with resources with Read operations." 
U don`t need transactions here at all, so just open session it try-with-resources

File: src/main/java/mate/academy/hibernate/relations/service/impl/CountryServiceImpl.java [hunk 3]
```diff
@@ -1,16 +1,28 @@
...
 import mate.academy.hibernate.relations.model.Country;
 import mate.academy.hibernate.relations.service.CountryService;
+import org.hibernate.SessionFactory;
 
 public class CountryServiceImpl implements CountryService {
+    private final CountryDao countryDao;
+
+    public CountryServiceImpl(SessionFactory sessionFactory) {
+        this.countryDao = new CountryDaoImpl(sessionFactory);
+    }
```
Comment (keda-vlad) [inline src/main/java/mate/academy/hibernate/relations/service/impl/CountryServiceImpl.java:15]: better to take into constructor argument CountryDao, create an implementation of this in main and add it to this constructor

File: src/main/java/mate/academy/hibernate/relations/service/impl/MovieServiceImpl.java [hunk 4]
```diff
@@ -1,16 +1,28 @@
...
 import mate.academy.hibernate.relations.model.Movie;
 import mate.academy.hibernate.relations.service.MovieService;
+import org.hibernate.SessionFactory;
 
 public class MovieServiceImpl implements MovieService {
+    private final MovieDao movieDao;
+
+    public MovieServiceImpl(SessionFactory sessionFactory) {
+        this.movieDao = new MovieDaoImpl(sessionFactory);
+    }
```
Comment (keda-vlad) [inline src/main/java/mate/academy/hibernate/relations/service/impl/MovieServiceImpl.java:15]: better to take into constructor argument MovieDao, create an implementation of this in main and add it to this constructor

**Review by keda-vlad — CHANGES_REQUESTED** [review]

**Review by keda-vlad — APPROVED** [review]

File: src/main/java/mate/academy/hibernate/relations/model/Actor.java [hunk 5]
```diff
@@ -1,8 +1,20 @@
...
+import javax.persistence.Table;
+
+@Entity
+@Table(name = "actors")
 public class Actor implements Cloneable {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
     private Long id;
     private String name;
+    @ManyToOne
```
Comment (nacenik) [inline src/main/java/mate/academy/hibernate/relations/model/Actor.java:17]: add `JoinTable` annotation  or `myppedBy`

File: src/main/java/mate/academy/hibernate/relations/model/Movie.java [hunk 6]
```diff
@@ -2,10 +2,22 @@
...
+import javax.persistence.Table;
 
+@Entity
+@Table(name = "movies")
 public class Movie implements Cloneable {
+    @Id
+    @GeneratedValue (strategy = GenerationType.IDENTITY)
     private Long id;
     private String title;
+    @ManyToMany(fetch = FetchType.EAGER)
```
Comment (nacenik) [inline src/main/java/mate/academy/hibernate/relations/model/Movie.java:20]: describe join table too

File: src/main/resources/hibernate.cfg.xml [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+        <property name="connection.username">root</property>
+        <property name="connection.password">vjzctvmz33</property>
+        <property name="show_sql">true</property>
+        <property name="hbm2ddl.auto">create-drop</property>
+
+        <mapping class="mate.academy.hibernate.relations.model.Actor"/>
+        <mapping class="mate.academy.hibernate.relations.model.Country"/>
+        <mapping class="mate.academy.hibernate.relations.model.Movie"/>
+    </session-factory>
+</hibernate-configuration>
```
Comment (nacenik) [inline src/main/resources/hibernate.cfg.xml:15]: add an empty line at the end of the file

**Review by nacenik — CHANGES_REQUESTED** [review]
Good solution, let's improve it :)

File: src/main/java/mate/academy/hibernate/relations/model/Movie.java [hunk 8]
```diff
@@ -2,10 +2,24 @@
...
+    @JoinTable
```
Comment (nacenik) [inline src/main/java/mate/academy/hibernate/relations/model/Movie.java:22]: Describe mapping. It's not enough to put this annotation

File: src/main/java/mate/academy/hibernate/relations/model/Actor.java [hunk 9]
```diff
@@ -1,8 +1,22 @@
...
+    @JoinColumn
```
Comment (nacenik) [inline src/main/java/mate/academy/hibernate/relations/model/Actor.java:19]: What the name of column for mapping?

**Review by nacenik — CHANGES_REQUESTED** [review]
Let's fix comments

**Review by andrii-hoienko — APPROVED** [review]
Good job!

**Pull request closed without merging**